lib/mauth/rack.rb

```diff
@@ -70,11 +86,20 @@ def should_authenticate?(env)
       end
     end
 
-    # signs outgoing responses
+    # signs outgoing responses with only the protocol used to sign the request.
     class ResponseSigner < MAuth::Middleware
       def call(env)
+        # QUESTION does instaniate a new request object, rather than doing a check for the headers here, add too much overhead?
+        versioned_token = MAuth::Rack::Request.new(env).signature_token
```

Comment:
We shouldn't need to rebuild the request object here.  Instead, I believe we can just stick something into the env in the `RequestAuthenticator` middleware to let us know here if we are on V1 or V2.